hmt_escrow/storage.py

```diff
@@ -70,9 +70,49 @@ def download(key: str, private_key: bytes) -> Dict:
     msg = _decrypt(private_key, ciphertext)
     return json.loads(msg)
 
+def createNewIpnsLink(name: str) -> str:
+    """Creates a new IPFS Id. The IPNS links are managed by key value system.
+       Pass in the key (e.g. 'intermediate-results-' + self.job_contract.address), 
+       then creats the ipns link
 
+    >>> import random 
+    >>> keyName = str(random.getrandbits(32 * 8)) # get random, or else throws duplicate key error
+    >>> createNewIpnsLink(keyName) != ''
+    True
+
+    Args:
+        name (str): Name to call the ipns link. Retrieve the link w/ same name
+
+    Returns:
+        str: Returns the IPNS url
+    """
+    name = name.lower()
+    key = IPFS_CLIENT.key.gen(name, 'ed25519')
+    return getIpnsLink(name)
+
+def getIpnsLink(name: str) -> str:
+    """Get the ipns link with the name of it which we remember it by
+
+    >>> import random 
+    >>> keyName = str(random.getrandbits(32 * 8)) # get random, or else throws duplicate key error
+    >>> createNewIpnsLink(keyName) != ''
+    True
+
+    Args:
+        name (str): Name we call ipns link
+
+    Returns:
+        str: Returns the IPNS url
+    """
+    keys = IPFS_CLIENT.key.list()
+    try:
+        return 'https://ipfs.io/ipns/' + list(filter(lambda x: x['Name'] == name.lower(), keys['Keys']))[0]['Id']
```

Comment:
do we want to hard code this ipfs url? might be better as an env var.  

Is this the only way users would look up the link address?

Reply:
I would use an env variable.

Also the oneliner is a bit unclear. I would maybe use a couple of lines there myself.

Reply:
One example of not using the 'https://...' is publishing the IPNS address offline. I don't know any other counter examples, but it is probably a good idea to add an ENV var.